src/go/mongolib/util/util.go

```diff
@@ -92,7 +96,7 @@ func GetReplicasetMembers(ctx context.Context, clientOptions *options.ClientOpti
 			membersMap[m.Name] = m
 		}
 
-		client.Disconnect(ctx)
+		client.Disconnect(ctx) //nolint
```

Comment:
Looks like we don't disconnect in case we get error.

Reply:
https://github.com/percona/percona-toolkit/pull/446/files#diff-a17f19d09995b36b3c78662622d66165R81
https://github.com/percona/percona-toolkit/pull/446/files#diff-a17f19d09995b36b3c78662622d66165R77

Reply:
We can extract it to separate task and fix it there.

Reply:
Yes please. It is in a for-loop where a new client is being created for each host found. 
The whole for-loop should be extracted into a new function to make it more clear and it will be possible to use a defer.

Reply:
Created https://jira.percona.com/browse/PT-1855 to make the improvements